pandas/core/indexes/datetimelike.py

```diff
@@ -83,12 +115,15 @@ def _join_i8_wrapper(joinf, dtype, with_indexers=True):
         """
         Create the join wrapper methods.
         """
+        from pandas.core.arrays.datetimelike import DatetimeLikeArrayMixin
 
         @staticmethod
         def wrapper(left, right):
-            if isinstance(left, (np.ndarray, ABCIndex, ABCSeries)):
+            if isinstance(left, (np.ndarray, ABCIndex, ABCSeries,
```

Comment:
shouldn't this be ExtensionArray ?

Reply:
In place of DateimeLikeArrayMixin? I don't think so. `.view` isn't part of the EA interface.